BUILD

```diff
@@ -2,23 +2,8 @@ package(default_visibility = ["//visibility:public"])
 
 cc_library(
     name = "libhoth",
-    srcs = [
-        "libhoth.c",
-        "libhoth_mtd.c",
-        "libhoth_spi.c",
-        "libhoth_usb.c",
-        "libhoth_usb_fifo.c",
-        "libhoth_usb_mailbox.c",
-    ],
-    hdrs = [
```

Comment:
This change (removing headers from `:libhoth`) might break downstream integrations of libhoth, who will have to include their desired subset of {`:libhoth_mtd`, `:libhoth_spi`, `:libhoth_usb`} for the backends they need.  This is probably necessary eventually, but I this we want to have this change include `deps = ["//:libhoth_mtd", "//:libhoth_spi", "//:libhoth_usb"],` here to avoid this churn for the moment.

Reply:
Added deps on `:libhoth_mtd`, `:libhoth_spi`, and `:libhoth_usb`.

Reply:
Update: removed these deps, as it would introduce a cyclic build dependency.